src/mono/mono/mini/interp/transform.c

```diff
@@ -4827,6 +4836,91 @@ interp_handle_box_patterns (TransformData *td, MonoClass *box_class, const unsig
 		td->ip = next_ip + 5;
 		return TRUE;
 	}
+	if (m_class_is_byreflike (box_class)) {
+		if (*next_ip == CEE_BRTRUE || *next_ip == CEE_BRTRUE_S || *next_ip == CEE_BRFALSE || *next_ip == CEE_BRFALSE_S) {
+			// replace
+			//  box ByRefLike
+			//  [brtrue/brfalse]
+			//
+			// by
+			//
+			// ldc.i4.s 1
+			// [brtrue/brfalse]
+			td->sp--;
+			interp_add_ins (td, MINT_LDC_I4_S);
+			td->last_ins->data[0] = (guint16) 1;
+			push_simple_type (td, STACK_TYPE_I4);
+			interp_ins_set_dreg (td->last_ins, td->sp [-1].var);
```

Comment:
Is the `[-1]` a comment mechanic in the intepreter? is there a macro or some other abstraction over that? I'm assuming the `sp` is "stack pointer", so perhaps that is the previous slot?

Reply:
Yea, I'm not sure why we do it this way.  But the general pattern is to decrement SP by the number of operands that an opcode expects (so now they're at `sp[0]` ... `sp[k-1]`) and use those `var`s, then do a push for the result(s) (which creates new vars) and then use `sp[-1]` to refer to the result.